### ISSUE TITLE
Customizable Context is needed

### ISSUE THREAD

**Issue by windymelt** [typelevel/feral#502]
I wrote AWS Lambda handler using Feral to handle IoT device (not Amazon IoT Button, but [SORACOM's one](https://www.soracom.io/iot-button/)) click event.

Since my IoT provider directly invokes lambda, lambda receives completely different `Context` from environment (for example it contains `simId` field). Thus runtime throws `Cannot read properties of undefined (reading 'installationId')` when decoding `Context`. When I invoke the lambda manually, the error does not occur.

Unfortunately, I cannot avoid this bahavior because decoding `Context` is completely untouchable.

![image](https://github.com/user-attachments/assets/826ebb5d-3cb1-4f2f-8ecf-a8cc0e76cbce)

In my use-case I don't need `Context` but input.

Would you mind to make `Context` type overridable and fully customizable?

## Appendix

For example SORACOM's context is described here: https://users.soracom.io/ja-jp/docs/funk/format/#nodejs-%E3%83%A9%E3%83%B3%E3%82%BF%E3%82%A4%E3%83%A0%E3%81%AE%E5%A0%B4%E5%90%88

**Comment by armanbilge** [comment]
> Since my IoT provider directly invokes lambda

Huh, can you explain more about what this means? Does it mean your lambda is not hosted on AWS servers at all?

**Comment by windymelt** [comment]
Sorry, my wording wasn't clear enough. Lambdas are hosted on my account on AWS servers as usual. I meant that lambdas aren't invoked through AWS API Gateway. It is invoked across my AWS account and provider account (by attaching policy).

**Comment by armanbilge** [comment]
I see, thanks for clarifying 🤔 is there any documentation about this? How do the official AWS Lambda libraries support custom contexts? On JVM and JS Feral is just a wrapper around the official libraries.

**Comment by bpholt** [comment]
Is this the `ClientContext` returned by the `Context.getClientContext` method? https://docs.aws.amazon.com/lambda/latest/dg/java-context.html

**Comment by bpholt** [comment]
I wonder if `clientContext.client` is `undefined` on https://github.com/typelevel/feral/blob/07b05570ee86ce1cfdd4204c94338ec0bb1de722/lambda/js/src/main/scala/feral/lambda/ContextPlatform.scala#L45

**Comment by armanbilge** [comment]
Yes, thanks, I think you're right! We probably need to model `client` it as an `UndefOr` in the facade.

https://github.com/typelevel/feral/blob/07b05570ee86ce1cfdd4204c94338ec0bb1de722/lambda/js/src/main/scala/feral/lambda/facade/Context.scala#L39-L43
